Pass `resources.limits.cpu` as `CRIBL_K8S_CPU_LIMIT` envvar

pdugas opened the pull request:
Goes along with CRIBL-30331 which honors this in the product to constrain the number of worker processes in Stream.  Setting it in Leader and Edge too for consistency.

michalbiesek commented on helm-chart-sources/logstream-leader/templates/_pod.tpl:
```diff
@@ -93,6 +93,12 @@ containers:
       - name: CRIBL_VOLUME_DIR
         value: {{ .Values.config.criblHome }}/config-volume
+      # CPU Limit
+      - name: CRIBL_KUBE_CPU_LIMIT
```
@pdugas, question
Kubernetes supports syntax such as '500m'
(see https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu).
I'm not sure if or how Kubernetes translates this syntax (if yes if this consistent behavior ?). If not, does this syntax would work as you expected, e.g. if 
CRIBL_KUBE_CPU_LIMIT would be set to 500m

pdugas replied:
@michalbiesek The product already has code to handle this.  See that PR.

michalbiesek reviewed (approved):
LGTM

benji-york commented on helm-chart-sources/edge/templates/daemonset.yaml:
```diff
@@ -105,6 +105,12 @@ spec:
+            - name: CRIBL_KUBE_CPU_LIMIT
+              valueFrom:
+                resourceFieldRef:
+                  container: {{ .Chart.Name }}
```
Do we expect the container name to always match the chart name?

pdugas replied:
it sets the container name further up in the spec using the same thing but it seems to work without specifying the container name there at all so I'm inclined to just remove it.

pdugas commented:
As it stands now, the default divisor is such that value in the envvar is rounded up to the nearest whole integer so setting the limit to `500m` or `0.5` results in `1` as the envvar value.  We could set the divisor to `1m` to get `500m` in this case instead and perhaps change the envvar name to `..._MCPU_LIMIT` to make it clear.

pdugas commented:
* Changed the name of the envvar from `..._KUBE_...` to `..._K8S_...` to match the pod name we already add.
* Added `divisor` so we get millicores without rounding instead of integer cores.

michalbiesek commented:
@pdugas
1. Please resolve conflicts
2. See https://github.com/pdugas/helm-charts/pull/1 (can be used directly or as an inspiration)
